Tarea 2

windie-rozureido opened the pull request:
Hola he subido la tarea 2 y el módulo de diferenciación automática.

windie-rozureido committed:
2-Ejercicios_AD.ipynb

windie-rozureido committed:
AutDiff.jl

lbenet commented:
> Las gráficas que se obtuvieron son representaciones geométricas discretas de los fractales, pero también lo es cualquier gráfica que se haga en computadora. Así que hablar de estos "fractales" tiene tando sentido como hablar de, por decir algo, la gráfica de sin(x).

Yo **no** coincido con este comentario; de hecho, la idea de hacer varias ampliaciones tiene como objeto ver que esto se trata de un fractal. Si uno hace varias ampliaciones de la función `sin(x)` uno termina por ver una línea continua. En este caso no. Parte del ejercicio es darse cuenta de que, algo tan común y corriente como es el método de Newton, de hecho, puede llevar a cosas interesantes!

lbenet commented:
>  y en otros es 3 órdenes de magnitud mayor que la épsilon de 1 de la máquina.

Lo que está escrito es correcto; sin embargo, `eps(868.84)` es `1.1368683772161603e-13` que es el error que estás obteniendo en el cálculo. O sea, es el número menor que le puedes sumar a 868.84 para obtener algo distinto.

lbenet commented on tareas/windie-rozureido/AutDiff.jl:
```diff
@@ -0,0 +1,64 @@
+end
+
+/(u::Dual,a::Real) = u/dual_num(a)
+/(a::Real,u::Dual) = /(u::Dual,a::Real)
```
Esta línea creo que te da problemas, ya que `a/u` lo convierte en `u/a`, cosa que **no** dará buenos resultados. Tal vez por esto es que hay líos en dos de las funciones donde encuentras problemas.

lbenet commented:
Sugiero corregir el error en la división, y ver si esto impacta en la función complicada en que checaste la derivada, para ver si es ahí de donde vienen los problemas. Además, para el ejercicio 1, sugiero hacer un ampliamiento, simplemente para saber (o tener alguna esperanza) que la frontera entre las regiones que convergen a una u otra raíz es suave en algún sentido.

windie-rozureido committed:
Cambios a a tarea.

Se corrigió el módulo de los Duales, la función de la división estaba mal definida.

windie-rozureido committed:
Merge branch 'master' of https://github.com/lbenet/2016-1_TemasSelectosFisComp

lbenet commented:
Yo veo bien la tarea; @UriAceves , tú qué opinas?

UriAceves commented:
> Las gráficas que se obtuvieron son representaciones geométricas discretas de los fractales, pero 
> también lo es cualquier gráfica que se haga en computadora.

Ya te lo habían comentado. NO, cualquier gráfica que se hace en una computadora no es un fractal. 

> frac·tal
> ˈfraktəl/
> Mathematics
> noun
> noun: fractal; plural noun: fractals
> 
>    1.
>    a curve or geometric figure, each part of which has the same statistical character as the whole...

Si tú me demuestras que cada gráfica generada por computadora cumple con la definición de fractal hago el merge; si no, sería bueno que corrigieras eso.

Lo demás lo veo bien.

lbenet commented:
Good catch!

windie-rozureido committed:
Se mejoró la redacción.

Se corrigió la redacción sobre las representaciones discretas y los fractales.

UriAceves commented:
Gerardo..., ¿cierto? Si modificas el archivo deja un comentario o algo por fa. Si no no nos damos cuenta tan fácil.
